core/src/main/java/com/imaginea/kodebeagle/GitHelper/ListFileContents.java

```diff
@@ -0,0 +1,432 @@
+			error(e, repository, "{0} failed to retrieve {1}", refs);
+		}
+		return list;
+	}
+
+	/**
+	 * Log an error message and exception.
+	 *
+	 * @param t
+	 * @param repository
+	 *            if repository is not null it MUST be the {0} parameter in the
+	 *            pattern.
+	 * @param pattern
+	 * @param objects
+	 */
+
+	private static void error(Throwable t, Repository repository, String pattern, Object... objects) {
+		List<Object> parameters = new ArrayList<Object>();
+		if (objects != null && objects.length > 0) {
+			for (Object o : objects) {
+				parameters.add(o);
+			}
+		}
+		if (repository != null) {
+			parameters.add(0, repository.getDirectory().getAbsolutePath());
+		}
+	}
+
+	/**
+	 * Returns the default branch to use for a repository. Normally returns
+	 * whatever branch HEAD points to, but if HEAD points to nothing it returns
+	 * the most recently updated branch.
+	 *
+	 * @param repository
+	 * @return the objectid of a branch
+	 * @throws Exception
+	 */
+	public static ObjectId getDefaultBranch(Repository repository) throws Exception {
+		ObjectId object = repository.resolve(Constants.HEAD);
+		if (object == null) {
+			// no HEAD
+			// perhaps non-standard repository, try local branches
+			List<RefModel> branchModels = getLocalBranches(repository, true, -1);
+			if (branchModels.size() > 0) {
+				// use most recently updated branch
+				RefModel branch = null;
+				Date lastDate = new Date(0);
+				for (RefModel branchModel : branchModels) {
+					if (branchModel.getDate().after(lastDate)) {
+						branch = branchModel;
+						lastDate = branch.getDate();
+					}
+				}
+				object = branch.getReferencedObjectId();
+			}
+		}
+		return object;
+	}
+
+	public static boolean isPossibleFilestoreItem(long size) {
+		return ((size >= 125) && (size <= 146));
+	}
+
+	/**
+	 * 
+	 * @return Representative FilestoreModel if valid, otherwise null
+	 */
+	public static FilestoreModel getFilestoreItem(ObjectLoader obj) {
+		try {
+			final byte[] blob = obj.getCachedBytes(146);
+			final String meta = new String(blob, "UTF-8");
+
+			return FilestoreModel.fromMetaString(meta);
+
+		} catch (LargeObjectException e) {
+			// Intentionally failing silent
+		} catch (Exception e) {
+			error(e, null, "failed to retrieve filestoreItem " + obj.toString());
+		}
+
+		return null;
+	}
+
+	/**
+	 * Returns the list of files changed in a specified commit. If the
+	 * repository does not exist or is empty, an empty list is returned.
+	 *
+	 * @param repository
+	 * @param commit
+	 *            if null, HEAD is assumed.
+	 * @param calculateDiffStat
+	 *            if true, each PathChangeModel will have insertions/deletions
+	 * @return list of files changed in a commit
+	 */
+
+	public static List<PathChangeModel> getFilesInCommit(Repository repository, RevCommit commit,
+			boolean calculateDiffStat) throws IOException {
+
+		List<PathChangeModel> list = new ArrayList<>();
+
+		if (!hasCommits(repository)) {
```

Comment:
Why are we checking it here? Should it not be done pretty upfront somewhere in "walkAllCommits" or something?